pkg/providers/nutanix/template.go

```diff
@@ -116,7 +116,10 @@ func (ntb *TemplateBuilder) GenerateEKSASpecSecret(clusterSpec *cluster.Spec, bu
 // EKSASecretName returns the name of the secret containing the credentials for the nutanix prism central and is used by the
 // EKS-Anywhere controller.
 func EKSASecretName(spec *cluster.Spec) string {
-	return spec.NutanixDatacenter.Spec.CredentialRef.Name
+	if spec.NutanixDatacenter.Spec.CredentialRef != nil {
```

Comment:
Does this adhere to the behavior described in https://github.com/aws/eks-anywhere/pull/5092#discussion_r1126935945?

Reply:
@chrisdoherty4 My understanding of that comment was restricted to not having a defaulting webhook for new cluster NutanixDatacenterConfig specs. We do have a reconcile for older NutanixDatacenterConfig objects that will add those fields to default values to ensure that the field is set everywhere.

Reply:
Just to clarify, we are not creating defaults for new clusters. If new cluster specs don't have a credentialRef they will still fail early with validation failures.
```
❌ Validation failed	{"validation": "nutanix Provider setup is valid", "error": "failed to validate cluster spec: credentialRef must be provided", "remediation": ""}
```

Reply:
Gotcha. Thanks for clarifying.